autorest/codegen/__init__.py

```diff
@@ -19,7 +19,7 @@
 
 def _get_credential_default_policy_type_has_async_version(credential_default_policy_type: str) -> bool:
     mapping = {
-        "BearerTokenCredentialPolicy": True,
+        "ARMChallengeAuthenticationPolicy": True,
```

Comment:
hey @msyyc, since we're trying to not make breaking changes, this change would be breaking for anyone who explictly passes in `BearerTokenCredentialPolicy`.

I realize mgmt plane wants to switch completely over to `ARMChallengeAuthenticationPolicy`, and since we check the authentication policy also in the templates, one idea is to have a `@property` on the `CodeModel`, i.e. `challenge_auth_policy`

This getter would replace every single `ARMChallengeAuthenticationPolicy that you currently have, and should have the following behavior
1. If users explicitly pass in `--credential-default-policy-type=BearerTokenCredentialPolicy`, it returns `BearerTokenCredentialPolicy`
2. If users are generating mgmt plane (i.e. `code_model.options['azure_arm'] == True`, return `ARMChallengeAuthenticationPolicy`.
3. Else, return `BearerTokenCredentialPolicy`

Reply:
Hi @iscai-msft Thank you for the detailed advice. 
(1) One little question: 
![image](https://user-images.githubusercontent.com/70930885/124711349-8c49c080-df30-11eb-9a6c-99561013d9da.png)
When users pass `BearerTokenCredentialPolicy`, the policy should not be replaced. So it should not bring breaking change : 
![image](https://user-images.githubusercontent.com/70930885/124711605-d7fc6a00-df30-11eb-9f9e-2071a3dbdd52.png)
Could you tell me more about it?
(2) By comparing `ARMChallengeAuthenticationPolicy` with `BearerTokenCredentialPolicy` (https://github.com/Azure/azure-sdk-for-python/blob/16f66ce00af0435b05bd7b88e045ba8ff42736ed/sdk/core/azure-mgmt-core/azure/mgmt/core/policies/_authentication.py#L36), we could find that the only difference is the `on_challenge` and it will also return 'false' if service does not support CAE. 

In one word, for users who don't input policy, `ARMChallengeAuthenticationPolicy` almost has same behavior with `BearerTokenCredentialPolicy`. So to simplify code, how about change the default policy directly?

Reply:
Yeah, but `ARMChallengeAuthenticationPolicy` is arm specific, and it's in `azure-mgmt-core`. So it makes sense to switch the default auth policy in the case of mgmt plane to `ARMChallengeAuthenticationPolicy`. However, for data-plane, we don't have users install `azure-mgmt-core` at all, so this is breaking for data-plane SDKs

Reply:
Got it. For mgmt-plan, the default policy could become `ARMChallengeAuthenticationPolicy`; for data-plan, the dafault policy keeps same with before. How about this commit : https://github.com/Azure/autorest.python/pull/966/commits/5d2821e402140be27a1d46977288b5c3dc81bd22